cms/static/cms/js/modules/cms.structureboard.js

```diff
@@ -1295,16 +1298,108 @@ class StructureBoard {
+
+        new_scripts.each(function() {
+            let script_exists = false;
+            let new_script = $(this);
+
+            if (new_script.hasClass('cms-execute-js-to-render')) {
+                $(old_scripts).each(function() {
+                    let old_script = $(this);
+
+                    if (new_script.prop('outerHTML') === old_script.prop('outerHTML')) {
+                        script_exists = true;
+                        return false;
+                    }
+                });
+                if (!script_exists) {
+                    let classList = new_script.attr('class').split(' ');
+
+                    classListCollection = classListCollection.concat(classList);
+                    if (typeof new_script.prop('src') === 'string' && new_script.prop('src') !== '') {
+                        scriptSrcList.push(new_script.prop('src'));
+                    } else {
+                        let js_file = document.createElement('script');
+
+                        js_file.textContent = new_script.prop('textContent') || '';
+                        js_file.type = 'text/javascript';
+                        document.body.appendChild(js_file);
+                    }
+                }
+            }
+        });
+        if (scriptSrcList.length === 0) {
+            that.triggerLoadEventsByClass(classListCollection);
+        } else {
+            that.getMultiScripts(scriptSrcList).done(function() {
+                that.triggerLoadEventsByClass(classListCollection);
+            });
+        }
+    }
+
+    /**
+     * Downloads and executes all scripts given by `scriptSrcList` and returns a
+     * `Promise` when all are done. This is to prevent multiple triggering of
+     * load events by `triggerLoadEventsByClass` in `addJsScriptsNeededForRender`.
+     * Original solution from https://stackoverflow.com/a/11803418
+     *
+     * @param {array} scriptSrcList   Array of script sources
+     *
+     * @returns {Promise}
+     */
+    getMultiScripts(scriptSrcList) {
+        let _arr = $.map(scriptSrcList, function(scr) {
+            return $.getScript(scr);
+        });
+
+        // eslint-disable-next-line new-cap
+        _arr.push($.Deferred(function(deferred) {
+            $(deferred.resolve);
+        }));
+
+        return $.when.apply($, _arr);
+    }
+
+    /**
+     * Triggers events if specific classes were in any of the scripts added by
+     * the method addJsScriptsNeededForRender
+     *
+     * @param {object} classListCollection  array of all classes the script tags had
+     */
+    triggerLoadEventsByClass(classListCollection) {
+        if (classListCollection.indexOf('cms-trigger-event-document-DOMContentLoaded') > -1) {
+            Helpers._getWindow().document.dispatchEvent(new Event('DOMContentLoaded'));
+        }
+        if (classListCollection.indexOf('cms-trigger-event-window-DOMContentLoaded') > -1) {
+            Helpers._getWindow().dispatchEvent(new Event('DOMContentLoaded'));
```

Comment:
personally i really doubt this will be of much use and would prefer to avoid it, since listening to this on window is quite rare

Reply:
Since the events can be turned on and off by the choice of the plugin developer, I would leave it in. Just in case someone needs it in an edge case, where i.e. a CDN included code triggers on this event. If no one uses it the only downside is some unused lines of code, but if someone needs it it a game changer.

Reply:
fair enough 👍